File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/examples/bossbar/BossbarExampleCreator.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers.examples.bossbar;

import com.gmail.borlandlp.minigamesdtools.arena.gui.providers.GUIProvider;
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.Creator;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo;

import java.util.ArrayList;
import java.util.List;

@CreatorInfo(creatorId = "bossbar_example")
public class BossbarExampleCreator extends Creator {
    @Override
    public GUIProvider create(String id, AbstractDataProvider dataProvider) throws Exception {
        return new BossbarExample();
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/examples/bossbar/BossbarExampleCreator.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers.examples.bossbar

import com.gmail.borlandlp.minigamesdtools.arena.gui.providers.GUIProvider
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.Creator
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo

@CreatorInfo(creatorId = "bossbar_example")
class BossbarExampleCreator : Creator() {
    @Throws(Exception::class)
    override fun create(id: String, dataProvider: AbstractDataProvider): GUIProvider {
        return BossbarExample()
    }
}